### PR TITLE
Delete Contract.* methods that are equivalent to Debug.Assert

### PR DESCRIPTION
**Pull request by jasonmalinowski**
These can be very confusing, since other methods on Contract actually throw exceptions. Since we have endless amounts of code directly calling Debug.Assert, let's just use it everywhere.

### CODE REVIEW

File: src/EditorFeatures/CSharp/Formatting/Indentation/CSharpIndentationService.Indenter.cs [hunk 1]
```diff
@@ -162,7 +163,7 @@ internal class Indenter : AbstractIndenter
                 if (token.IsSemicolonOfEmbeddedStatement() ||
                     token.IsCloseBraceOfEmbeddedBlock())
                 {
-                    Contract.Requires(
+                    Debug.Assert(
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/CSharp/Formatting/Indentation/CSharpIndentationService.Indenter.cs]: this file itself confuses me.  We had Contract.Assert and Contract.Requires.  And both mapped to Debug.Assert.  Does that mean Contract itself had two methods that were identical just with different names?  Yikes...

File: src/Workspaces/Core/Portable/Utilities/Contract.cs [hunk 2]
```diff
@@ -7,60 +7,6 @@ namespace Roslyn.Utilities
...
-            }
-        }
-
-        /// <summary>
-        /// Equivalent to Debug.Assert.  
-        ///
-        /// DevDiv 867813 covers removing this completely at a future date
-        /// </summary>
-        [Conditional("DEBUG")]
-        public static void Assume(bool condition, string message = null)
```
Comment (CyrusNajmabadi) [inline src/Workspaces/Core/Portable/Utilities/Contract.cs]: i literally have no words.  3 methods in a row that all are Equivalent to Debug.Assert...

File: src/EditorFeatures/CSharp/Formatting/Indentation/CSharpIndentationService.Indenter.cs (hunk 1 above)
Comment (jasonmalinowski) [reply src/EditorFeatures/CSharp/Formatting/Indentation/CSharpIndentationService.Indenter.cs]: Hence my desire to remove the confusion!

**Comment by heejaechang** [comment]
is Contract.Requires(expression) and Debug.Assert(expression) compiled differently in debug and release?

I just merged this and got build error since expression doesn't exist in release. but Contract.Requires built fine.

**Comment by jasonmalinowski** [comment]
@heejaechang Both were [Conditional("DEBUG")]. Something tells me you have some #if DEBUG going on somewhere else.

**Comment by heejaechang** [comment]
yep. I saw that as well. just want to check whether there is some other compiler magic going on. I removed the code anyway, so fine now.

**Pull request merged**